server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2358,7 +2358,8 @@ long getNumDocUpdates() {
         }
         Searcher searcher = acquireSearcher(source, SearcherScope.INTERNAL);
         try {
-            LuceneChangesSnapshot snapshot = new LuceneChangesSnapshot(searcher, mapperService, minSeqNo, maxSeqNo, requiredFullRange);
+            final int batchSize = preferredSearchBatchSize <= 0 ? Engine.LUCENE_HISTORY_DEFAULT_SEARCH_BATCH_SIZE : preferredSearchBatchSize;
```

Comment:
I think it is better to define this default just in ccr (`ShardChanges.java`)?
(and let newLuceneChangesSnapshot(...) methods just not allow values lower than 1)

Reply:
If we do this (and I'm not convinced we should) that we please not use a magic number for the parameter? people should just say what they want or use this constant directly